### DISCUSSION TITLE
0.10.0-rc2 has been released!

### DISCUSSION THREAD

**Discussion by mihe** [godot-jolt/godot-jolt#669] [Announcements]
This is the second **release candidate** for version 0.10.0, and assuming no major bugs/regressions are found is likely to become 0.10.0-stable.

This release supports [**Godot 4.1**](https://github.com/godotengine/godot/releases/tag/4.1.3-stable).

## Changelog

These are the notable changes that have been made since [0.9.0-stable](https://github.com/godot-jolt/godot-jolt/releases/tag/v0.9.0-stable) was released. You can also find a list of all the commits [here](https://github.com/godot-jolt/godot-jolt/compare/v0.9.0-stable...v0.10.0-rc2).

Any breaking changes are denoted with ⚠️.

### Changed

- ⚠️ Changed so that ray-casts using the `hit_from_inside` parameter report a zero normal when hitting a convex shape from inside, to match Godot Physics.
- Changed the `space_get_contacts` method of `PhysicsServer3D` (and thus also the "Visible Collision Shapes" debug rendering) to no longer include contacts generated by overlaps with `Area3D`.

### Added

- Added support for Android (ARM64, ARM32, x86-64 and x86).
- Added support for iOS.

### Fixed

- Fixed issue where an error saying `Parameter "body" is null` would be emitted after freeing certain bodies while they were in contact with a `CharacterBody3D`.

<hr /><em>This discussion was created from the release <a href='https://github.com/godot-jolt/godot-jolt/releases/tag/v0.10.0-rc2'>0.10.0-rc2</a>.</em>

**Comment by mihe** [comment]
Again, to anyone stumbling onto this release from somewhere else: I'd be very curious to hear if/how this release works on your Android/iOS device, especially more specialized Android devices like the various VR headsets out there.

**Reply by phelioz** [reply]
The new optimized build seems to fixed the heavy stuttering on my Quest 2, the physics runs smooth for me now, nice!
Smooth on my Quest 3 too :)
